docs/content/start/telemetry.md

```diff
@@ -0,0 +1,175 @@
+
+```ts
+/**
+ * Base interface for logging telemetry statements.
+ * Can contain any number of properties that get serialized as json payload.
+ * @param category - category of the event, like "error", "performance", "generic", etc.
+ * @param eventName - name of the event.
+ */
+export interface ITelemetryBaseEvent extends ITelemetryProperties {
+  category: string;
+  eventName: string;
+}
+```
+
+### Category
+
+Currently, there are 3 categories used by the telemetry system:
+
+- errors - used to classify known errors, e.g. duplicate data store IDs.
+- performance - used to track metrics, e.g. used by the summarizer to track how long it takes to do or load a summary (if it's takes too long then even though its a performance event it'll be considered an error).
+- generic - used as a catchall for events that are mostly harmless.
+
+### EventName
+
+This property is currently used by the telemetry system to indicate the event in a more descriptive manner.
+
+### Adding complexity
+
+Similar to the `ITelemetryBaseLogger` interface mentioned above, different levels of event complexity can also be achieved by adding other attributes to the object implementing the `ITelemetryBaseEvent` interface. Below are some examples:
+
+```ts
+if (chunk.version !== undefined) {
+  logger.send({
+      eventName: "MergeTreeChunk:serializeAsMinSupportedVersion",
+      category: "generic",
+      fromChunkVersion: chunk.version,
+      toChunkVersion: undefined,
+  });
+}
+```
+
+The code snippet here implements a telemetry event without adding much complexity. The telemetry event object here adds only a few other attributes to the `ITelemetryBaseEvent`.
+
+```ts
+/**
+ * Send a telemetry event with the logger
+ *
+ * @param event - the event to send
+ * @param error - optional error object to log
+ */
+public sendTelemetryEvent(event: ITelemetryGenericEvent, error?: any) {
+  const newEvent: ITelemetryBaseEvent = {
+    ...event,
+    category: event.category ?? (error === undefined ?  "generic" : "error"),
+  };
+  if (error !== undefined) {
+    TelemetryLogger.prepareErrorObject(newEvent, error, false);
+  }
+  this.send(newEvent);
+}
+```
+
+However, the code snippet here shows a telemetry event object that adds much more complex logics to determine the event that has occurred, then passed the resulting event to the `send()` method for the container's telemetry system to output.
+
+Here is the above telemetry event object in action:
+
+```ts
+// Adding this event temporarily so that we can get help debugging if something goes wrong.
```

Comment:
should remove this comment since it's not relevant to the example

Reply:
removed! 😃